`read_path` and `write_path` fail when `credentials` attribute is used

<!-- Thanks for submitting an issue! Please describe the issue. -->

# Expectation / Proposal
I expect to be able to use the `credentials` field along with the `write_path` and `read_path` methods. When I use `credentials`, I get the following error: ` ValueError: S3 Bucket requires either a minio_credentialsfield or an aws_credentials field.`

# Traceback / Example
```
def _get_bucket_resource(self) -> boto3.resource:
        """
        Retrieves boto3 resource object for the configured bucket
        """
        if self.minio_credentials:
            params_override = (
                self.minio_credentials.aws_client_parameters.get_params_override()
            )
            if "endpoint_url" not in params_override and self.endpoint_url:
                params_override["endpoint_url"] = self.endpoint_url
            bucket = (
                self.minio_credentials.get_boto3_session()
                .resource("s3", **params_override)
                .Bucket(self.bucket_name)
            )
    
        elif self.aws_credentials:
            params_override = (
                self.aws_credentials.aws_client_parameters.get_params_override()
            )
            bucket = (
                self.aws_credentials.get_boto3_session()
                .resource("s3", **params_override)
                .Bucket(self.bucket_name)
            )
        else:
>           raise ValueError(
                "S3 Bucket requires either a minio_credentials"
                "field or an aws_credentials field."
            )
E           ValueError: S3 Bucket requires either a minio_credentialsfield or an aws_credentials field.
```

- [x] I would like to [help contribute](https://PrefectHQ.github.io/prefect-aws/#contributing) a pull request to resolve this!

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/pycqa/isort
-    rev: 5.10.1
+    rev: 5.12.0
     hooks:
       - id: isort
         language_version: python3

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fixed errors raised when using `write_path` and `read_path` with `credentials` field on `S3Bucket` - [#208](https://github.com/PrefectHQ/prefect-aws/pull/208)
+
 ## 0.2.4
 
 Released on January 23rd, 2023.

--- a/prefect_aws/credentials.py
+++ b/prefect_aws/credentials.py
@@ -38,6 +38,7 @@ class AwsCredentials(CredentialsBlock):
 
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
     _block_type_name = "AWS Credentials"
+    _documentation_url = "https://prefecthq.github.io/prefect-aws/credentials/#prefect_aws.credentials.AwsCredentials"  # noqa
 
     aws_access_key_id: Optional[str] = Field(
         default=None,
@@ -166,6 +167,7 @@ class MinIOCredentials(CredentialsBlock):
         "docs: https://docs.min.io/docs/minio-server-configuration-guide.html "
         "for more info about the possible credential configurations."
     )
+    _documentation_url = "https://prefecthq.github.io/prefect-aws/credentials/#prefect_aws.credentials.MinIOCredentials"  # noqa
 
     minio_root_user: str = Field(default=..., description="Admin or root user.")
     minio_root_password: SecretStr = Field(

--- a/prefect_aws/ecs.py
+++ b/prefect_aws/ecs.py
@@ -269,6 +269,9 @@ class ECSTask(Infrastructure):
     _block_type_name = "ECS Task"
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
     _description = "Run a command as an ECS task."  # noqa
+    _documentation_url = (
+        "https://prefecthq.github.io/prefect-aws/ecs/#prefect_aws.ecs.ECSTask"  # noqa
+    )
 
     type: Literal["ecs-task"] = Field(
         "ecs-task", description="The slug for this task type."

--- a/prefect_aws/s3.py
+++ b/prefect_aws/s3.py
@@ -256,6 +256,9 @@ class S3Bucket(WritableFileSystem, WritableDeploymentStorage, ObjectStorageBlock
 
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/1jbV4lceHOjGgunX15lUwT/db88e184d727f721575aeb054a37e277/aws.png?h=250"  # noqa
     _block_type_name = "S3 Bucket"
+    _documentation_url = (
+        "https://prefecthq.github.io/prefect-aws/s3/#prefect_aws.s3.S3Bucket"  # noqa
+    )
 
     bucket_name: str = Field(default=..., description="Name of your bucket.")
 
@@ -418,7 +421,9 @@ class S3Bucket(WritableFileSystem, WritableDeploymentStorage, ObjectStorageBlock
         Authenticate MinIO credentials or AWS credentials and return an S3 client.
         This is a helper function called by read_path() or write_path().
         """
-        if self.minio_credentials:
+        if self.credentials:
+            s3_client = self.credentials.get_s3_client()
+        elif self.minio_credentials:
             s3_client = self.minio_credentials.get_s3_client()
 
         elif self.aws_credentials:
@@ -434,7 +439,18 @@ class S3Bucket(WritableFileSystem, WritableDeploymentStorage, ObjectStorageBlock
         """
         Retrieves boto3 resource object for the configured bucket
         """
-        if self.minio_credentials:
+        if self.credentials:
+            params_override = (
+                self.credentials.aws_client_parameters.get_params_override()
+            )
+            if "endpoint_url" not in params_override and self.endpoint_url:
+                params_override["endpoint_url"] = self.endpoint_url
+            bucket = (
+                self.credentials.get_boto3_session()
+                .resource("s3", **params_override)
+                .Bucket(self.bucket_name)
+            )
+        elif self.minio_credentials:
             params_override = (
                 self.minio_credentials.aws_client_parameters.get_params_override()
             )

--- a/tests/test_s3.py
+++ b/tests/test_s3.py
@@ -315,9 +315,9 @@ def s3_bucket(s3, request, aws_creds_block, minio_creds_block):
     key = request.param
 
     if key == "aws_credentials":
-        fs = S3Bucket(bucket_name=BUCKET_NAME, aws_credentials=aws_creds_block)
+        fs = S3Bucket(bucket_name=BUCKET_NAME, credentials=aws_creds_block)
     elif key == "minio_credentials":
-        fs = S3Bucket(bucket_name=BUCKET_NAME, minio_credentials=minio_creds_block)
+        fs = S3Bucket(bucket_name=BUCKET_NAME, credentials=minio_creds_block)
 
     s3.create_bucket(Bucket=BUCKET_NAME)
 

